File: 234_Palindrome_Linked_List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        vector<int> x; 
        if(head == NULL) return true; 
        while(head)
        {
            x.push_back(head->val); 
            head = head->next; 
        }
        
        int left = 0, right = x.size() - 1; 
        while(left < right)
        {
            if(x[left++] != x[right--]) return false; 
        }
        return true;
    }
};

File: 234_Palindrome_Linked_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: ListNode) -> bool:
        listOfValues = []
        while head != None:
            listOfValues.append(head.val)
            head = head.next 
            
        left = 0
        right = len(listOfValues) - 1 
        while(left < right):
            if(listOfValues[left] != listOfValues[right]):
                return False 
            left += 1 
            right -= 1 
        return True 
        